src/pages/api/secure/group/full.ts

```diff
@@ -12,14 +13,35 @@ const handle = async (req: NextApiRequest, res: NextApiResponse) => {
       return res.status(401).send('Unauthorized')
     }
     try {
-      const groups = await getGroupsAndMembers(
+      const allGroups = await getGroupsAndMembers(
         account_address,
         extractQuery(req.query, 'limit'),
         extractQuery(req.query, 'offset'),
         extractQuery(req.query, 'search'),
         extractQuery(req.query, 'includeInvites') === 'true'
       )
-      return res.status(200).json(groups)
+
+      // Check subscription status for filtering
+      const isPro = await isProAccountAsync(account_address)
+
+      if (!isPro) {
+        // Free tier: return only first 5 groups
+        const limitedGroups = allGroups.slice(0, 5)
```

Comment:
Filtering for non-pro accounts should be handled on the server-side after the fixed number of resources has been pulled(See the comment above).